cpp/src/column/column_factories.cpp

```diff
@@ -81,4 +85,50 @@ std::unique_ptr<column> make_fixed_width_column(
   return make_numeric_column(type, size, state, stream, mr);  
 }
 
+struct column_from_scalar_dispatch 
+{
+  template <typename T>
+  std::enable_if_t<cudf::is_fixed_width<T>(), std::unique_ptr<cudf::column>>
+  operator()( scalar const& value, size_type size,
+              rmm::mr::device_memory_resource* mr, cudaStream_t stream)
+  {
+    auto output_column = make_fixed_width_column(value.type(),size,mask_state::UNALLOCATED,stream,mr);
+    auto view = output_column->mutable_view();
+    experimental::detail::fill_in_place(view,0,size,value,stream);
+    return output_column;
+  }
+
+  template <typename T>
+  std::enable_if_t<std::is_same<cudf::string_view, T>::value, std::unique_ptr<cudf::column>>
+  operator()( scalar const& value, size_type size,
+              rmm::mr::device_memory_resource* mr, cudaStream_t stream)
+  {
+    auto p_scalar = static_cast<experimental::scalar_type_t<T> const*>(&value);
+    auto null_mask = create_null_mask(size, mask_state::ALL_NULL, stream, mr);
+    column_view sc{ data_type{STRING}, size, nullptr, 
+                    static_cast<bitmask_type*>(null_mask.data()), size };
+    auto output = cudf::strings::detail::fill(cudf::strings_column_view(sc),
+                                              0, size, *p_scalar, mr, stream);
+    output->set_null_mask(rmm::device_buffer{},0);
+    return output;
+  }
+  
+  template <typename T>
+  std::enable_if_t<std::is_same<cudf::dictionary32, T>::value, std::unique_ptr<cudf::column>>
+  operator()( scalar const& value, size_type size,
+              rmm::mr::device_memory_resource* mr, cudaStream_t stream)
+  {
+    CUDF_FAIL("dictionary not supported when creating from scalar");
+  }
+};
+
+std::unique_ptr<column> make_column_from_scalar(scalar const& s, size_type size,
+    rmm::mr::device_memory_resource* mr, cudaStream_t stream )
+{
+  CUDF_EXPECTS( s.is_valid(), "scalar parameter must be valid");
```

Comment:
`fill` can handle filling with null scalars, yes? If so, might as well let people make a column from a null scalar.

Reply:
Creating an all-null column is already possible with the other factory methods and does not require a call to `fill()` so I thought I would leave that out of this function. 
I can add that to the CUDF_EXPECTS string if you want. Or it is easy enough to support here too as well -- just check the flag and don't call `fill()`.

Reply:
I'm thinking of cases where the scalar `s` is the result of some function like `reduce` where the user may not know whether or not it is null without explicitly checking. It's just nice for functions to handle situations like this so callers don't have to explicitly check for it.